libdrgn/debug_info.c

```diff
@@ -1535,28 +1581,41 @@ parse_member(struct drgn_debug_info *dbinfo, Dwarf_Die *die,
 		bit_field_size = 0;
 	}
 
-	struct drgn_lazy_parameter member_type;
-	struct drgn_error *err = drgn_lazy_type_from_dwarf(dbinfo, die,
-							   can_be_incomplete_array, bias,
-							   "DW_TAG_member",
-							   &member_type);
+	struct drgn_lazy_parameter member;
+	if (dwarf_tag(die) == DW_TAG_subprogram) {
+		// Creating objects
+		err = drgn_lazy_object_from_dwarf(dbinfo, die, bias, name, &member);
+		if (err)
+			return err;
+
+		err = drgn_compound_type_builder_add_member(builder, member, name,
+							    0, bit_field_size);
+		if (err)
+			goto err;
+		return NULL;
```

Comment:
Could you refactor this so you only need to call `drgn_compound_type_builder_add_member()` in one place?